### PR TITLE
feat(sast): sast original semgrep basic runner

### PR DESCRIPTION
**Pull request by achiar99**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

[//]: # "
    # PR Title
    Be aware that we use the title to create changelog automatically and therefore only allow specific prefixes
    - break:    to indicate a breaking change, this supersedes any of the types
    - feat:     to indicate new features or checks
    - fix:      to indicate a bugfix or handling of edge cases of existing checks
    - docs:     to indicate an update to our documentation
    - chore:    to indicate adjustments to workflow files or dependency updates
    - platform: to indicate a change needed for the platform
    Additionally a scope is needs to be added to the prefix, which indicates the targeted framework, in doubt choose 'general'.
    #    
    Allowed prefixs:
    ansible|argo|arm|azure|bicep|bitbucket|circleci|cloudformation|dockerfile|github|gha|gitlab|helm|kubernetes|kustomize|openapi|sca|secrets|serverless|terraform|general|graph|terraform_plan
    #
    ex.
    feat(terraform): add CKV_AWS_123 to ensure that VPC Endpoint Service is configured for Manual Acceptance
"

Implement basic sast runner based on original semgrep module

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my feature, policy, or fix is effective and works
- [ ] New and existing tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by arielkru — APPROVED** [review]
👑

File: checkov/common/sast/enums.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
+
```
Comment (rotemavni) [inline checkov/common/sast/enums.py:1]: @achiar99 Why not place this file under `checkov/sast`?

File: checkov/common/sast/enums.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+from enum import Enum
+
+
+class ExtendedEnum(Enum):
+    @classmethod
+    def list(cls):
+        return list(map(lambda c: c.value, cls))
```
Comment (rotemavni) [inline checkov/common/sast/enums.py:8]: Why do we need this extended enum? Wouldn't it be enough to implement `list(cls)` in `SastLanguages`?

File: checkov/sast/checks/rules/python/SuperuserPort.yaml [hunk 3]
```diff
@@ -0,0 +1,27 @@
+rules:
+  - id: CKV_SAST_1
+    patterns:
+      - pattern: set_port($ARG)
+      - metavariable-comparison:
+          metavariable: $ARG
+          comparison: $ARG < 1024
+    message: module setting superuser port
+    languages:
+      - python
```
Comment (rotemavni) [inline checkov/sast/checks/rules/python/SuperuserPort.yaml:10]: Given the fact that this check is under `..rules/python/..`, if we want to implement this for TS would we need to duplicate this to `..rules/ts/` and just change the language?

File: checkov/sast/runner.py [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+         all_targets,
+         renamed_targets,
+         target_manager_ignore_log,
+         filtered_rules,
+         profiler,
+         profiling_data,
+         parsing_data,
+         explanations,
+         shown_severities,
+         target_manager_lockfile_scan_info) = run_semgrep(output_handler=output_handler, target=targets,
```
Comment (rotemavni) [inline checkov/sast/runner.py:97]: seems like `run_semgrep` returns much more data, shouldn't we explicitly ignore the ones we don't need, so we won't accidentally get weird values for the ones we do need (in terms of the param order)
image

File: checkov/sast/runner.py [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+                                                          pattern="", lang="", configs=config, **{})
+        semgrep_output = SemgrepOutput(filtered_matches_by_rule, semgrep_errors, all_targets, renamed_targets,
```
Comment (rotemavni) [inline checkov/sast/runner.py:99]: Nice!

File: checkov/sast/runner.py [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+                                       target_manager_ignore_log, filtered_rules, profiler, profiling_data,
+                                       parsing_data, explanations, shown_severities, target_manager_lockfile_scan_info)
+        return semgrep_output
+
+    def _get_report(self, filtered_matches_by_rule: Dict[Rule, List[RuleMatch]]) -> Report:
```
Comment (rotemavni) [inline checkov/sast/runner.py:104]: I prefer `_create_report` since this method is actually creating the report.

File: checkov/sast/runner.py [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+                file_abs_path = match.match.location.path
+                file_path = file_abs_path.split('/')[-1]
+                severity = get_severity(SEMGREP_SEVERITY_TO_CHECKOV_SEVERITY.get(rule.severity))
+                file_line_range = [match.start.line, match.end.line]
+                check_result = {'result': CheckResult.FAILED, 'evaluated_keys': {}}
+
+                record = Record(check_id=check_id, check_name=check_name, resource=None, evaluations={},
+                                check_class=None, check_result=check_result, code_block=code_block,
+                                file_path=file_path, file_line_range=file_line_range,
+                                file_abs_path=file_abs_path, severity=severity)
```
Comment (rotemavni) [inline checkov/sast/runner.py:120]:
```suggestion
                check_result = _CheckResult(result=CheckResult.FAILED)

                record = Record(check_id=check_id, check_name=check_name, resource="", evaluations={},
                                check_class="", check_result=check_result, code_block=code_block,
                                file_path=file_path, file_line_range=file_line_range,
                                file_abs_path=file_abs_path, severity=severity)
```

Had some typing mismatches

File: checkov/sast/checks/rules/python/SuperuserPort.yaml (hunk 3 above)
Comment (achiar99) [reply checkov/sast/checks/rules/python/SuperuserPort.yaml:10]: I think its going to be rare to have the exactly rule for multiple languages.
IMO in that case its ok to duplicate

**Pull request closed without merging**